dlog/commitment/src/commitment.rs

```diff
@@ -75,9 +77,22 @@ where
     }
 }
 
-pub fn shift_scalar<F: PrimeField>(x: F) -> F {
-    let two: F = 2_u64.into();
-    x - two.pow(&[F::Params::MODULUS_BITS as u64])
+pub fn shift_scalar<G: AffineCurve>(x: G::ScalarField) -> G::ScalarField
```

Comment:
can you add some doc here?

Reply:
Probably it would be sufficient to summarise [this](https://github.com/MinaProtocol/mina/blob/39dd86401bb56ee06a6004fe8ac4f952050d4a6d/src/lib/pickles_types/shifted_value.ml#L34) and [this](https://github.com/MinaProtocol/mina/blob/39dd86401bb56ee06a6004fe8ac4f952050d4a6d/src/lib/pickles_types/shifted_value.ml#L136) comment.

Reply:
not sure I understand the second comment, why is the constant 5 used? And shouldn't the result of the scalar mul also be split between high bit and low bits?

Reply:
Added a comment